src/repo_type.py

```diff
@@ -49,13 +49,18 @@ def getHead(self):
         '''
         Gets the latest version or commit for this software
         Needs to be implemented by extending class.
-        
+
         Returns String
         '''
         raise NotImplementedError
-    
-    
-    
+
+    def checkout(self,commit):
+        '''
+        checks out to the given commit
+        '''
+
+        return self.repo_ref.checkout(commit)
```

Comment:
As this is an Abstract class (python doesn't natively support it *grumble*) and can be extended by RepoTypes that are not git based, I don't think this implementation is generic enough.

That is to say we are still planning on supporting tar/rpm/etc ?

Suggest just to put in a "raise NotImplementedError" here to simulate a method that needs to be implemented by the extending class